### PR TITLE
Add support for enrolling Snowflake data sources

### PR DESCRIPTION
**Pull request by zturechek**
Adds support for the Snowflake handler and metadata. Secure views can be enabled for enrolled data sources.

Tested on Immuta app v2021.1.2

### CODE REVIEW

**Review by kaypeter87 — APPROVED** [review]
LGTM. Nice small code that will add a really important snowflake integration.

File: fh_immuta_utils/client.py [hunk 1]
```diff
@@ -103,6 +103,10 @@ def make_glob_request_headers(cls, config: Dict[str, Any]) -> Dict[str, Any]:
             return cls.make_generic_odbc_request_headers(config)
         if config["handler_type"] == "Amazon Athena":
             return cls.make_athena_glob_request_headers(config)
+        if config["handler_type"] == "Snowflake":
+            headers = cls.make_generic_odbc_request_headers(config)
+            headers["sql-warehouse"] = config["warehouse"]
+            return headers
```
Comment (briantist) [inline fh_immuta_utils/client.py:109]: it's a bit strange that codecov is calling these lines out as not being covered isn't it? I see tests added, is it possible this code path is still not being reached by the tests?

**Review by chintanms — APPROVED** [review]

File: fh_immuta_utils/client.py (hunk 1 above)
Comment (zturechek) [reply fh_immuta_utils/client.py:109]: Correct, it is not being reached. Historically I'm not sure why tests don't exist. `client.py` is made up mostly of the functions that make the API requests. This class instantiates the client. We'll have to get some mocking going here to test this.

Comment (zturechek) [reply fh_immuta_utils/client.py:109]: Clarifying that once I'm back, I'll add the test coverage for this.

Comment (zturechek) [reply fh_immuta_utils/client.py:109]: Tests added!

Comment (briantist) [reply fh_immuta_utils/client.py:109]: lgtm!

**Pull request merged**